lib/iris/experimental/regrid.py

```diff
@@ -953,10 +953,10 @@ def _src_align_and_flatten(coord):
         min_sx, min_tx = np.min(sx.points), np.min(tx.points)
         if min_sx < 0 and min_tx >= 0:
             indices = np.where(sx_points < 0)
-            sx_points[indices] += modulus
+            sx_points[indices] = sx_points[indices] + modulus
```

Comment:
Yes, it will upcast. Previously, int the case where the dtype of the array was int and `modulus` was a float, it would simply round `modulus` down to the nearest integer, which is clearly wrong.